broker/amqp.go

```diff
@@ -30,16 +32,18 @@ func NewAMQP(group string, subgroup string, receiveCallback func(string, []byte)
 
 // Connect will connect this client to the AQMP Server
 func (a *AMQP) Connect(url string) error {
-	conn, err := amqp.Dial(url)
+	pubConn, err := amqp.Dial(url)
 	if err != nil {
 		return err
 	}
-	a.conn = conn
-	ch, err := conn.Channel()
+	a.publishConn = pubConn
+    	consumeConn, err := amqp.Dial(url)
```

Comment:
Not a fan of having this here. We should only establish a consumer connection when the broker is being used to consume.

Reply:
If that's what you would like to do, I could move that into the subscribe() function instead.

Reply:
Alrighty, I have moved it.

Now that the consumer connection is optional, I think that two separate close() methods would be warranted? One for publisher, one for consumer.

Reply:
I'm not sure what the issue is to check for a nil connection in the Close function and I can't see any usecase where one might want to close only one. Ideally unsubscribing from all queues should close the connection automatically.

Reply:
Alright. I've added that revision, see https://github.com/spec-tacles/go/pull/4/commits/fa36ef9305546e5c981b15d245ecd1f7a7aa9392.